### PR TITLE
OBPIH-3527 Create Invoice wizard with first page

### PR DESCRIPTION
**Pull request by awalkowiak**

### CODE REVIEW

File: grails-app/domain/org/pih/warehouse/invoice/Invoice.groovy [hunk 1]
```diff
@@ -84,4 +89,37 @@ class Invoice implements Serializable {
...
+                if (referenceNumber.referenceNumberType == referenceNumberType) {
+                    return referenceNumber
+                }
+            }
+        }
+        return null
+    }
+
+    ReferenceNumber getVendorInvoiceNumber() {
+        return getReferenceNumber("Vendor Invoice Number")
```
Comment (jmiranda) [inline grails-app/domain/org/pih/warehouse/invoice/Invoice.groovy:106]: Replace "Vendor Invoice Number" with a constant.
```
Constants.VENDOR_INVOICE_NUMBER_ID = "VENDOR_INVOICE_NUMBER"
```

File: grails-app/migrations/0.8.x/changelog-2021-03-03-1330-create-table-invoice.xml [hunk 2]
```diff
@@ -114,32 +114,9 @@
...
     </preConditions>
     <createTable tableName="invoice_reference_number">
-      <column name="invoice_id" type="CHAR(38)"/>
+      <column name="invoice_reference_numbers_id" type="CHAR(38)"/>
       <column name="reference_number_id" type="CHAR(38)"/>
+      <column name="reference_numbers_idx" type="INT"/>
     </createTable>
   </changeSet>
-
-  <changeSet author="awalkowiak" id="0303202113300-8">
```
Comment (jmiranda) [inline grails-app/migrations/0.8.x/changelog-2021-03-03-1330-create-table-invoice.xml:122]: We still need these foreign key references. 

invoice_reference_number.invoice_reference_numbers_id -> invoice.id
invoice_reference_number.reference_number_id -> reference_number.id

File: grails-app/migrations/0.8.x/changelog-2021-03-16-1250-insert-reference-number-type-vendor-invoice-number.xml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog/1.9 http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-1.9.xsd">
+
+    <changeSet author="dhatala" id="1536634735965-2">
+        <preConditions onFail="MARK_RAN">
+            <sqlCheck expectedResult="0">
+                <![CDATA[SELECT count(*) FROM reference_number_type WHERE name="Vendor Invoice Number"]]>
+            </sqlCheck>
+        </preConditions>
+        <insert tableName="reference_number_type">
+            <column name="id" valueNumeric="11" />
```
Comment (jmiranda) [inline grails-app/migrations/0.8.x/changelog-2021-03-16-1250-insert-reference-number-type-vendor-invoice-number.xml:13]: Replace `11` with `VENDOR_INVOICE_NUMBER`

**Review by jmiranda — CHANGES_REQUESTED** [review]

**Comment by DominikaHatala** [comment]
@jmiranda I have applied required changes.

File: grails-app/migrations/0.8.x/changelog-2021-03-16-1250-insert-reference-number-type-vendor-invoice-number.xml (hunk 3 above)
Comment (DominikaHatala) [reply grails-app/migrations/0.8.x/changelog-2021-03-16-1250-insert-reference-number-type-vendor-invoice-number.xml]: @jmiranda I'm getting this error " Caused By: Unknown column 'VENDOR_INVOICE_NUMBER' in 'field list' after changing it

**Comment by DominikaHatala** [comment]
@jmiranda I have updated this PR with required changes.

**Review by jmiranda — APPROVED** [review]

File: src/js/actions/index.js [hunk 4]
```diff
@@ -77,6 +79,26 @@ export function fetchReasonCodes() {
...
+  const request = apiClient.get(url);
+
+  return {
+    type: FETCH_CURRENCIES,
+    payload: request,
+  };
+}
+
+export function fetchOrganizations() {
+  const url = '/openboxes/api/organizations';
```
Comment (awalkowiak) [inline src/js/actions/index.js:93]: @DominikaHatala, @jmiranda I think this should be debounced and we could return only org name and id. Currently, this request takes 2.5s and response has size of 1.6 MB (locally)

**Pull request merged**